src/libraries/System.Runtime.InteropServices.JavaScript/tests/JSImportGenerator.UnitTest/Compiles.cs

```diff
@@ -71,70 +71,118 @@ internal static partial void Annotated(object a1, long a2, long a3, global::Syst
                                     {
-                                        __delegate_arg_arg1.ToJS(__delegate_arg1);
-                                    });
-                                    __a4_native__js_arg.ToJS(a4);
-                                    __a3_native__js_arg.ToJSBig(a3);
-                                    __a2_native__js_arg.ToJS(a2);
-                                    __a1_native__js_arg.ToJS(a1);
-                                    global::System.Runtime.InteropServices.JavaScript.JSFunctionBinding.InvokeJS(__signature_Annotated_1583225186, __arguments_buffer);
+                                        global::System.Span<global::System.Runtime.InteropServices.JavaScript.JSMarshalerArgument> __arguments_buffer = stackalloc global::System.Runtime.InteropServices.JavaScript.JSMarshalerArgument[]
+                                        {
+                                            ____arg_exception_native,
+                                            ____arg_return_native,
+                                            __a1_native,
+                                            __a2_native,
+                                            __a3_native,
+                                            __a4_native,
+                                            __a5_native,
+                                            __a6_native,
+                                            __a7_native,
+                                            __a8_native,
+                                            __a9_native,
+                                            __a10_native,
+                                            __a11_native,
+                                            __a12_native,
+                                            __a13_native,
+                                            __a14_native,
+                                            __a15_native
+                                        };
+                                        global::System.Runtime.InteropServices.JavaScript.JSFunctionBinding.InvokeJS(__signature_Annotated_1583225186, __arguments_buffer);
+                                        return __arguments_buffer[1];
```

Comment:
This should be `void`

Reply:
This being non-void is how we get the return value out to pass back to managed code (since the return value has to be passed in to form the buffer.

Reply:
See line 151